src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java

```diff
@@ -121,7 +121,13 @@ public void start() {
         // Requires authentication endpoint without specific authenticator attached
         addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
 
-        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
+        // Direct basic auth authentication (web services)
+        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
```

Comment:
Hi,

I’m a bit confused as to how to test those last two endpoints. Please can you provide instructions for a dummy (assuming I do so from the web page, how do I do the post, for example, with a specified header?) How do I get a valid JWT token for testing without leaving a user logged in via an indirect client?

Cheers,

Jez


> On 29 Dec 2015, at 13:20, LELEU Jérôme <notifications@github.com> wrote:
> 
> In src/main/java/org/pac4j/vertx/verticle/DemoServerVerticle.java <https://github.com/pac4j/vertx-pac4j-demo/pull/35#discussion_r48539623>:
> 
> > @@ -121,7 +121,13 @@ public void start() {
> >          // Requires authentication endpoint without specific authenticator attached
> >          addProtectedEndpointWithoutAuthorizer("/protected/index.html", "", router);
> >  
> > -        router.get("/index.html").handler(DemoHandlers.indexHandler(config));
> > +        // Direct basic auth authentication (web services)
> > +        addProtectedEndpointWithoutAuthorizer("/dba/index.html", "DirectBasicAuthClient", router);
> /rest-jwt/index.html link:
> 
> If I click on the link, I get a "unauthorized" error: OK.
> If I had "?token=" in the url, I get an internal error: OK, this is the expected behaviour, but the error page is not the right one (I expected: error500.html).
> If I provide a good JWT, it works: the profile is properly displayed on the specific page. Though, when I come back to the index page, the profile is still displayed meaning it has been saved in the session while it should have only been available in the request (by default) for this kind of call.
> —
> Reply to this email directly or view it on GitHub <https://github.com/pac4j/vertx-pac4j-demo/pull/35/files#r48539623>.
>